service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -487,6 +508,25 @@ protected void doStop(final Future<Void> stopFuture) {
         return result;
     }
 
+    /**
+     * Checks if the number of connections exceeded the configured limit for the given tenant or not.
+     *
+     * @param tenantConfig The config of the tenant to be checked.
+     * @return If the limit is not exceeded, then a succeeded future containing the tenant configuration is returned.
+     *         Else a failed future with a {@link ClientErrorException} containing <em>403 Forbidden</em> status.
+     */
+    protected final Future<TenantObject> isConnectionLimitExceeded(final TenantObject tenantConfig) {
```

Comment:
I would rather have a default implementation being injected (in particular in the case of no Prometheus being available) instead of this method ...

Reply:
@sophokles73 Do you mean a default implementation of the `ResourceLimitChecks`? If Prometheus host name is not set, then inject this default implementation. Anyway the protocol adapters will invoke the `isConnectionLimitExceeded(...)` in `AbstractProtocolAdapterBase` class. If the prometheus host name is set or not, accordingly the default or the `PrometheusBasedResourceLimitChecks` implementation will be used.

Currently if no prometheus host name is set, then no implementation is being injected. Then `ResourceLimitChecks` instance will be null. If it is null, then simply `true` is returned when adapters invoke `isConnectionLimitExceeded(...)`. I try to understand the advantages that the default implementation brings.

Reply:
Using a default implementation will prevent us from having to add more and more methods with ugly null checks to `AbstractProtocolAdapterBase` when `ResourceLimitChecks` is extended with new methods. Instead, we only need the *getResourceLimitChecks()` method and return the instance.
By default, this would be the dummy implementation (which always passes all checks). The dummy implementation can then be overwritten by using the setter. This is the same approach we also use for the OpenTracing `Tracer` ...

Reply:
@sophokles73 I pushed my changes. Can you pls take a look?

Reply:
sorry, but I do not see any changes having been pushed since three days ...

Reply:
@sophokles73 Sorry, there was a mistake. Now it should be there.

Reply:
IMHO we should simply remove this method and let subclasses use *getResourceLimitChecks* directly instead of introducing more and more of these *delegating* methods which do not add value.